daffodil-test/src/test/resources/org/apache/daffodil/section05/facets/PatternRanges.tdml

```diff
@@ -0,0 +1,123 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  Licensed to the Apache Software Foundation (ASF) under one or more
+  contributor license agreements.  See the NOTICE file distributed with
+  this work for additional information regarding copyright ownership.
+  The ASF licenses this file to You under the Apache License, Version 2.0
+  (the "License"); you may not use this file except in compliance with
+  the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+-->
+<tdml:testSuite
+  xmlns:dfdl="http://www.ogf.org/dfdl/dfdl-1.0/"
+  xmlns:dfdlx="http://www.ogf.org/dfdl/dfdl-1.0/extensions"
+  xmlns:tdml="http://www.ibm.com/xmlns/dfdl/testData"
+  xmlns:ex="http://example.com"
+  xmlns:xs="http://www.w3.org/2001/XMLSchema"
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  defaultRoundTrip="onePass"
+  defaultValidation="on">
+
+  <tdml:defineSchema name="s">
+    <xs:include schemaLocation="org/apache/daffodil/xsd/DFDLGeneralFormat.dfdl.xsd"/>
+
+    <dfdl:format ref="ex:GeneralFormat"/>
+
+    <xs:simpleType name="singleByteCharsWithCodepointLessThan1F"
+                    dfdl:encoding="iso-8859-1">
+      <xs:restriction base="xs:string">
+        <!--
+        Using Daffodil's E000 remapping for the XML illegal characters from 00 to 08,
+        09 and 0A are legal, but we must use \t and \n for them since the pattern facet is an attribute value.
+        Using E000 remapping for XML illegal characters  0B and 0C
+        0D is legal (though remapped to LF), but we must use \r for that.
+        Using E000 remapping for XML illegal characters 0E to 1F
+        20 to 7F are legal.
+        -->
+        <!--
+          We cannot depend on Daffodil's mapping from E009 to 9, E00A to A, and E00D to D.
+          Because that won't work in Xerces with "full" validation.
+          We can't use numeric entities for the tab, LF, or CR, because those aren't allowed in attribute
+          values. So we use \t, \n, and \r for those control characters.
+          -->
+        <xs:pattern value="[&#xE000;-&#xE008;\t\n&#xE00B;&#xE00C;\r&#xE00E;-&#xE01F;&#x20;-&#x7F;]*"/>
+      </xs:restriction>
+    </xs:simpleType>
+
+
+    <xs:element name="str" type="ex:singleByteCharsWithCodepointLessThan1F"
+       dfdl:lengthKind="delimited"/>
```

Comment:
I was a bit puzzled at seeing dfdl:lengthKind `delimited` on a solitary root element until I read the DFDL spec and realized the end of the data stream delimits an element as well.  However, the spec says we also can use the dfdl:lengthKind `endOfParent` to allow the last element to consume data up to the end of the data stream; might `endOfParent` be more self-documenting too?

Reply:
Yes, but endOfParent isn't supported yet by Daffodil.

Reply:
I'll add a comment about this not having a delimiter, so uses end-of-file.